openedx/features/course_experience/__init__.py

```diff
@@ -22,49 +23,31 @@ class DefaultTrueWaffleFlagNamespace(WaffleFlagNamespace):
         and refactor/fix any tests that shouldn't be removed.
 
     """
-    def is_flag_active(self, flag_name, check_before_waffle_callback=None):
+    def _get_flag_active_request(self, namespaced_flag_name, request):
         """
-        Overrides is_flag_active, and returns and caches whether the provided flag is active.
-
-        If the flag value is already cached in the request, it is returned.
-        If the flag doesn't exist, always returns default of True.
-
-        Note: This is a similified version of the method it overrides, that
-        hard codes the default to True, and skips the call back used for
-        course overrides:
-        https://github.com/edx/edx-platform/blob/df9be8c678f8266e2e5710513c74deca14c4527c/openedx/core/djangoapps/waffle_utils/__init__.py#L229-L305
-
+        Override the default flag value in the absence of flag: if the
+        flag does not exist, it is assumed to be True (active). See the
+        parent class.
         """
         # Import is placed here to avoid model import at project startup.
         from waffle.models import Flag
 
-        # validate arguments
-        namespaced_flag_name = self._namespaced_name(flag_name)
-        value = self._cached_flags.get(namespaced_flag_name)
-        if value is None:
-
-            # determine if the flag is undefined in waffle
-            try:
-                Flag.objects.get(name=namespaced_flag_name)
-            except Flag.DoesNotExist:
-                # default to True if not defined
-                value = True
-
-            if value is None:
-                request = crum.get_current_request()
-                if request:
-                    value = flag_is_active(request, namespaced_flag_name)
-                else:
-                    set_custom_attribute('warn_flag_no_request', True)
-                    # Return the default value if not in a request context.
-                    # Same as the original implementation
-                    self._set_waffle_flag_attribute(namespaced_flag_name, value)
-                    return True
-
-            self._cached_flags[namespaced_flag_name] = value
-
-        self._set_waffle_flag_attribute(namespaced_flag_name, value)
-        return value
+        # determine if the flag is undefined in waffle
+        try:
+            Flag.objects.get(name=namespaced_flag_name)
+        except Flag.DoesNotExist:
+            # default to True if not defined
+            return True
+
+        return super()._get_flag_active_request(namespaced_flag_name, request)
+
+
+class CourseWaffleFlag(BaseCourseWaffleFlag):
+    """
+    Override the default flag values by ignoring course-specific behaviour.
+    """
+    def is_enabled_for_course(self, course_key):
+        return None
```

Comment:
This would have caused a backward-compatibility issue.
1. Since this is basically ignoring course overrides, let's not call this `CourseWaffleFlag`.  Maybe something like `DefaultTrueWaffleFlag`?  We should explain in the docstring that it is very temporary, and maybe have an `__init__` assertion that it only gets used with `DefaultTrueWaffleFlagNamespace`?
2. Below, you should only use this new class for CourseWaffleFlags that are using `DEFAULT_TRUE_WAFFLE_FLAG_NAMESPACE`.  The others, like `DISPLAY_COURSE_SOCK_FLAG`, should use the real CourseWaffleFlag, and should continue to behave normally.

Reply:
> This would have caused a backward-compatibility issue.

Right! Very good catch.

> Since this is basically ignoring course overrides, let's not call this CourseWaffleFlag. Maybe something like DefaultTrueWaffleFlag? We should explain in the docstring that it is very temporary, and maybe have an __init__ assertion that it only gets used with DefaultTrueWaffleFlagNamespace?

:ok_hand: 

> Below, you should only use this new class for CourseWaffleFlags that are using DEFAULT_TRUE_WAFFLE_FLAG_NAMESPACE. The others, like DISPLAY_COURSE_SOCK_FLAG, should use the real CourseWaffleFlag, and should continue to behave normally.

:ok_hand: